logrotate.c

```diff
@@ -2484,6 +2485,14 @@ static int writeState(const char *stateFilename)
             return 1;
         }
 
+        if (flock(fdcurr, LOCK_EX | LOCK_NB) == -1) {
```

Comment:
Why do we care about locking in `writeState()`?  Isn't it too late?

Some log files might have been already rotated at this point.  Why should we prevent the status file from being written in case we are not able to acquire the lock?

Reply:
This is just paranoid locking of the temporary state file, shouldn't matter much though.

Reply:
In which scenario exactly would the code help?  I still think that the state file should be written.  When the logs have already been rotated, not reflecting it in the state file could hardly improve anything.

Reply:
Changed to fail if state file does not exist at this point

Reply:
Looks good. Thanks for the update!